main.c

```diff
@@ -26,21 +26,37 @@ bool box_intersect(const struct slurp_box *a, const struct slurp_box *b) {
 		a->height + a->y > b->y;
 }
 
+struct slurp_selection *seat_get_current_selection(struct slurp_seat *seat) {
+	if (seat->pointer_selection.has_selection) {
+		return &seat->pointer_selection;
+	} else if (seat->touch_selection.has_selection) {
+		return &seat->touch_selection;
+	} else {
+		return NULL;
+	}
+}
+
 static struct slurp_output *output_from_surface(struct slurp_state *state,
 	struct wl_surface *surface);
 
 static void move_seat(struct slurp_seat *seat, wl_fixed_t surface_x,
 		wl_fixed_t surface_y) {
-	int x = wl_fixed_to_int(surface_x) + seat->current_output->logical_geometry.x;
-	int y = wl_fixed_to_int(surface_y) + seat->current_output->logical_geometry.y;
+	struct slurp_selection *current_selection =
+		seat_get_current_selection(seat);
+	if (current_selection == NULL) {
+	  //return;
+	  current_selection = &seat->pointer_selection; //still need to track the mouse coordinates;
```

Comment:
Hmm, so if there's no current selection, why do we need to update the pointer selection?

Reply:
move_seat updates mouse coordinates which are the start of the selection when clicked. Otherwise the mouse selection will always start at 0,0, i.e. in the corner.

I know it's crutchy but right now I can't think of any other way.